engine/classes/ElggDiskFilestore.php

```diff
@@ -68,7 +68,7 @@ public function open(\ElggFile $file, $mode) {
 
 		$path = substr($fullname, 0, $ls);
 
-		if (($mode != 'write') && (!file_exists($fullname))) {
+		if (($mode === 'open') && (!file_exists($fullname))) {
```

Comment:
Oops... it should be `read`... the validation is done  in the switch a few lines below